scripts/generate-docs-types.js

```diff
@@ -0,0 +1,287 @@
+    return "default: `[]`";
+  }
+
+  switch (types) {
+    case "`boolean`":
+      return "default: `false`";
+
+    case "`string`":
+      return 'default: `""`';
+
+    default:
+      return "default: `null`";
+  }
+};
+
+const typeToDoc = type => line => {
+  const match = line.match(callRegex);
+  const id = match[1].slice(0, 1) === "_" ? match[1].slice(1) : match[1];
+  const required = match[2] === ":";
+  const types =
+    typeOverwrites.hasOwnProperty(type) &&
+    typeOverwrites[type].hasOwnProperty(id)
+      ? typeOverwrites[type][id]
+      : "`" + removeNullUndefined(match[3]) + "`";
+  return ` - \`${id}\`: ${types} (${getRequirement(
+    required,
+    id,
+    type,
+    types
+  )})`;
+};
+
+const seeAlso = type => {
+  const see = { assert: "", is: "" };
+  const is = String(`is${type}`).toLowerCase();
+  const assert = String(`assert${type}`).toLocaleLowerCase();
+  for (const type in t) {
+    if (type.toLowerCase() === assert) {
+      see.assert = type;
+    }
+    if (type.toLowerCase() === is) {
+      see.is = type;
+    }
+  }
+  return `See also \`t.${see.is}(node, opts)\` and \`t.${
+    see.assert
+  }(node, opts)\`.`;
+};
+
+const findAliases = typeDefLines => {
+  const aliasBlock = [];
+  let collect = false;
+  for (const line of typeDefLines) {
+    if (line === "export interface Aliases {") {
+      collect = true;
+    } else if (collect && line === "}") {
+      collect = false;
+    } else if (collect) {
+      const parts = line.split(":");
+      aliasBlock.push(parts[0].trim());
+    }
+  }
+  const aliases = {};
+  aliasBlock.forEach(alias => {
+    const types = typeDefLines.filter(l =>
+      new RegExp(`^export type ${alias} = `).test(l)
+    );
+    const names = types[0]
+      .slice(`export type ${alias} = `.length)
+      .split(" | ")
+      .map(s => (s.slice(-1) === ";" ? s.slice(0, -1) : s));
+    for (const name of names) {
+      if (aliases.hasOwnProperty(name)) {
+        aliases[name].push(alias);
+      } else {
+        aliases[name] = [alias];
+      }
+    }
+  });
+  for (const key in aliases) {
+    aliases[key].sort((a, b) => a.localeCompare(b));
+  }
+  return aliases;
+};
+
+const getArgumentsAndReturnType = def => {
+  const match = def.match(signatureRegex);
+  if (match && match[1] && match[2]) {
+    return [match[1].split(",").map(s => String(s).trim()), match[2]];
+  } else if (def.includes("()")) {
+    const r = def.match(typeRegex);
+    if (r && r[1]) {
+      return [[], r[1]];
+    }
+  }
+  return [[], "unknown"];
+};
+
+(async () => {
+  const typeDefs = await readfile(
+    path.resolve(
+      __dirname,
+      "node_modules",
+      "@babel",
+      "types",
+      "lib",
+      "index.d.ts"
+    ),
+    "utf-8"
+  );
+  const typeDefLines = typeDefs.split("\n");
+  const aliases = findAliases(typeDefLines);
+  const sections = [];
+
+  for (const type in t) {
+    const isBuilder = /_builder\.default/.test(t[type].toString());
```

Comment:
I think we should include not just the builder, but as well as `assert*` and `is*` utility functions